src/include/txn.i

```diff
@@ -297,10 +302,40 @@ __wt_txn_modify(WT_SESSION_IMPL *session, WT_UPDATE *upd)
 	op->type = F_ISSET(session, WT_SESSION_LOGGING_INMEM) ?
 	    WT_TXN_OP_INMEM : WT_TXN_OP_BASIC;
 #ifdef HAVE_TIMESTAMPS
-	if (__wt_txn_update_needs_timestamp(session, op))
+	if (__wt_txn_update_needs_timestamp(session, op)) {
 		__wt_timestamp_set(&upd->timestamp, &txn->commit_timestamp);
+
+		/*
+		 * Checkpoint transactions and lookaside transactions will not
+		 * be prepared.
+		 */
+		if (!WT_SESSION_IS_CHECKPOINT(session) &&
+		    ((WT_BTREE *)op->dhandle->handle)->id !=
+		    S2C(session)->cache->las_fileid) {
+			/*
+			 * Store the key, to search the update incase of
+			 * prepared transaction.
+			 */
+			if (btree->type == BTREE_ROW) {
```

Comment:
I'm sad that we need code that knows the difference between row and column stores here, but I don't see an obvious alternative.
@keithbostic Could you take a look and let us know if you could see an alternative?

Reply:
We could also skip metadata updates here as well, if that matters, since they aren't prepared either.